lldb/source/Symbol/CoreDumpOptions.cpp

```diff
@@ -0,0 +1,37 @@
+//===-- CoreDumpOptions.cpp -------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "lldb/Symbol/CoreDumpOptions.h"
+
+using namespace lldb;
+using namespace lldb_private;
+
+void CoreDumpOptions::SetCoreDumpPluginName(const llvm::StringRef name) {
+  m_core_dump_plugin_name = name.data();
+}
+
+std::optional<llvm::StringRef> CoreDumpOptions::GetCoreDumpPluginName() const {
```

Comment:
We should return a `std::optional<std::string>` to match the ivar unless call sites all want `llvm::StringRef`